common/src/chain/transaction/signature/inputsig/authorize_pubkey_spend.rs

```diff
@@ -72,200 +72,107 @@ mod test {
-            let witness = prepare_data_for_wrong_destination(sighash_type);
-            assert!(matches!(
-                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap_err(),
-                TransactionSigError::InvalidSignatureEncoding
-            ));
+    fn invalid_signature() {
+        let (private_key, public_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
+        let destination = Destination::PublicKey(public_key.clone());
+        let tx = generate_unsigned_tx(&destination, 1, 2).unwrap();
+
+        for sighash_type in sig_hash_types() {
+            let witness = StandardInputSignature::produce_signature_for_input(
+                &private_key,
+                sighash_type,
+                destination.clone(),
+                &tx,
+                INPUT_NUM,
+            )
+            .unwrap();
+            let mut raw_signature = witness.get_raw_signature().clone();
+            raw_signature[0] = raw_signature[0].wrapping_add(2);
+
+            assert_eq!(
+                Err(TransactionSigError::InvalidSignatureEncoding),
+                AuthorizedPublicKeySpend::from_data(&raw_signature),
+                "{sighash_type:X?}"
+            )
         }
     }
 
-    fn prepare_data_for_sign(sighash_type: SigHashType) -> (PrivateKey, PublicKey, H256) {
-        const INPUT_NUM: usize = 0;
+    #[test]
+    fn test_verify_public_key_spending() {
         let (private_key, public_key) = PrivateKey::new(KeyKind::RistrettoSchnorr);
-        let outpoint_dest = Destination::PublicKey(public_key.clone());
-        let tx = generate_unsigned_tx(outpoint_dest.clone()).unwrap();
-        let witness = StandardInputSignature::produce_signature_for_input(
-            &private_key,
-            sighash_type,
-            outpoint_dest,
-            &tx,
-            INPUT_NUM,
-        )
-        .unwrap();
-        let sighash = signature_hash(witness.sighash_type(), &tx, INPUT_NUM).unwrap();
-        (private_key, public_key, sighash)
+        let destination = Destination::PublicKey(public_key.clone());
+        let tx = generate_unsigned_tx(&destination, 1, 2).unwrap();
+
+        for sighash_type in sig_hash_types() {
+            let witness = StandardInputSignature::produce_signature_for_input(
+                &private_key,
+                sighash_type,
+                destination.clone(),
+                &tx,
+                INPUT_NUM,
+            )
+            .unwrap();
+            let spender_signature =
+                AuthorizedPublicKeySpend::from_data(witness.get_raw_signature()).unwrap();
+            let sighash = signature_hash(witness.sighash_type(), &tx, INPUT_NUM).unwrap();
+            verify_public_key_spending(&public_key, &spender_signature, &sighash)
+                .expect(&format!("{sighash_type:X?}"));
+        }
     }
 
     #[test]
     fn test_sign_pubkey_spending() {
```

Comment:
Isn't the verify test a superset of the sign test? Why are there two separate tests for that?

Reply:
Alrighty. Keep them both.